tests/WorkflowTest.php

```diff
@@ -36,6 +38,43 @@
     assertEquals(1, $workflow->refresh()->jobs_processed);
 });
 
+it('it stores finished job id, defaulting to class name', function () {
+    $job1 = new TestJob1();
+    $workflow = createWorkflow([
+        'job_count' => 1,
+        'jobs_processed' => 0,
+    ]);
+
+    $workflow->onStepFinished($job1);
+
+    assertEquals([$job1::class], $workflow->refresh()->finished_jobs);
+});
+
+it('it stores finished job id', function () {
+    $job1 = (new TestJob1())->withJobId('::job-id::');
+    $workflow = createWorkflow([
+        'job_count' => 1,
+        'jobs_processed' => 0,
+    ]);
+
+    $workflow->onStepFinished($job1);
+
+    assertEquals(['::job-id::'], $workflow->refresh()->finished_jobs);
+});
```

Comment:
I think these two tests should be the same test but using a data provider instead. Something like this (untested)

```php
it('stores a finished job\'s id', function ($job, string $expectedJobId) {
    $workflow = createWorkflow([
        'job_count' => 1,
        'jobs_processed' => 0,
    ]);

    $workflow->onStepFinished($job);

    assertEquals([$expectedJobId], $workflow->refresh()->finished_jobs);
})->with([
    'no job id should default to class name' => [
        new TestJob1(),
        TestJob1::class,
    ],
    'use existing job id' => [
        (new TestJob1())->withJobId('::job-id::'),
        '::job-id::',
    |
]);
```